reindexer/reindex_request_processor/src/main/scala/uk/ac/wellcome/platform/reindex/processor/services/ReindexWorkerService.scala

```diff
@@ -0,0 +1,51 @@
+package uk.ac.wellcome.platform.reindex.processor.services
+
+import akka.actor.{ActorSystem, Terminated}
+import com.google.inject.Inject
+import uk.ac.wellcome.messaging.sns.NotificationMessage
+import uk.ac.wellcome.messaging.sqs.SQSStream
+import uk.ac.wellcome.models.reindexer.{ReindexRequest, ReindexableRecord}
+import uk.ac.wellcome.storage.ObjectStore
+import uk.ac.wellcome.storage.dynamo._
+import uk.ac.wellcome.storage.vhs.{EmptyMetadata, VersionedHybridStore}
+import uk.ac.wellcome.utils.JsonUtil._
+
+import scala.concurrent.{ExecutionContextExecutor, Future}
+import scala.util.Try
+
+class ReindexWorkerService @Inject()(
+  versionedHybridStore: VersionedHybridStore[ReindexableRecord,
+                                             EmptyMetadata,
+                                             ObjectStore[ReindexableRecord]],
+  sqsStream: SQSStream[NotificationMessage],
+  system: ActorSystem) {
+
+  implicit val executionContext: ExecutionContextExecutor = system.dispatcher
+
+  sqsStream.foreach(this.getClass.getSimpleName, processMessage)
+
+  private def processMessage(message: NotificationMessage): Future[Unit] =
+    for {
+      _ <- Future.fromTry(
+        Try {
+          val reindexRequest = fromJson[ReindexRequest](message.Message).get
+          versionedHybridStore.updateRecord(reindexRequest.id)((
+            ReindexableRecord(reindexRequest.id, reindexRequest.desiredVersion),
+            EmptyMetadata()))(
```

Comment:
Maybe throw an error if we try to reindex a non-existent record? It should never happen, but “should” isn’t a strong guarantee.